examples/flax/image-captioning/run_image_captioning_flax.py

```diff
@@ -0,0 +1,1485 @@
+        # remove problematic examples
+        # (if feature extraction is performed at the beginning, the filtering is done during preprocessing below
+        # instead here.)
+        if not run_feat_ext_at_beginning:
+            predict_dataset = predict_dataset.filter(
+                filter_fn, batched=True, num_proc=data_args.preprocessing_num_workers
+            )
+        predict_dataset = predict_dataset.map(
+            function=function_kwarg,
+            batched=True,
+            num_proc=data_args.preprocessing_num_workers,
+            # kept image paths
+            remove_columns=remove_columns_kwarg,
+            load_from_cache_file=not data_args.overwrite_cache,
+            desc=f"Running {processor_names} on prediction dataset",
+            fn_kwargs={"max_target_length": data_args.val_max_target_length},
+            features=features_kwarg,
+        )
+        if run_feat_ext_at_beginning:
+            # set format (for performance) since the dataset is ready to be used
+            predict_dataset = predict_dataset.with_format("numpy")
+
+    # Store some constant
+
+    train_batch_size = int(training_args.per_device_train_batch_size) * jax.device_count()
+
+    if training_args.block_size % train_batch_size > 0:
+        raise ValueError(
+            f"`training_args.block_size` needs to be a multiple of the global batch size. Got {training_args.block_size} and {train_batch_size} instead."
+        )
+
+    if training_args.do_train:
+        steps_per_epoch = len(train_dataset) // train_batch_size
+        num_train_examples_per_epoch = steps_per_epoch * train_batch_size
+        num_epochs = int(training_args.num_train_epochs)
+        total_train_steps = steps_per_epoch * num_epochs
+    else:
+        num_train_examples_per_epoch = 0
+
+    eval_batch_size = int(training_args.per_device_eval_batch_size) * jax.device_count()
+
+    if training_args.do_eval:
+        num_eval_examples = len(eval_dataset)
+        eval_steps = num_eval_examples // eval_batch_size
+
+    if training_args.do_predict:
+        num_test_examples = len(predict_dataset)
+        test_steps = num_test_examples // eval_batch_size
+
+    def blockwise_data_loader(
+        rng: jax.random.PRNGKey,
+        ds: Dataset,
+        block_size: int,
+        batch_size: int,
+        shuffle: bool = False,
+        keep_in_memory: bool = False,
+        split: str = "",
+    ):
+        """
+        Wrap the simple `data_loader` in a block-wise way if `block_size` > 0, else it's the same as `data_loader`.
+
+        If `block_size` > 0, it requires `ds` to have a column that gives image paths in order to perform image feature
+        extraction (with the column name being specified by `image_column`). The tokenization should be done before
+        training in this case.
+        """
+
+        if shuffle:
+            indices = jax.random.permutation(rng, len(ds))
+            indices = np.asarray(indices)
```

Comment:
same comment as above, just use `jax.numpy` here

Reply:
@patil-suraj - what is the advantage of doing so?

Reply:
it might not cause any slowdown here, but IMO would be nice to be consistent, rather altering between `np` and `jnp`, to avoid confusion.

Reply:
Using JAX array to interact with `Datasets` will slow down a lot, and that's why I tried to use numpy array here.

The reason I first using jax array then converting numpy array it to keep as much as the code in `run_summarization_flax.py`, because it seems that JAX's PRNGs work differently from NumPy.

I can use only numpy array for this part, or just leave a comment about why converting from JAX to NumPy here to avoid confusion.

Reply:
I see. Thanks for the explanation. So let's just keep it as it is and leave a comment there about why we need to do `jnp` => `np`